app/decorators/user_decorator.rb

```diff
@@ -116,6 +116,17 @@ def devices?
     !recent_devices.empty?
   end
 
+  def last_signed_in_at
+    user.devices.order(last_used_at: :desc).first&.last_used_at
+  end
+
+  def second_last_signed_in_at
+    user.events.where(
+      event_type: ['sign_in_before_2fa',
+                   'sign_in_after_2fa'],
```

Comment:
If we're including `sign_in_after_2fa`, do we need to include `sign_in_before_2fa`?

Reply:
Yeah, I questioned that myself. As the method is named `second_last_signed_in_at` I figured I'd keep it generic as possible and count both events. Open to other suggestions.

I find the name `second_last_signed_in_at` somewhat awkward, but found it necessary for a currently signed in user. Maybe the wording can be changed. Otherwise a method like `last_signed_in_at` returns the current login date for the `current_user`.

Reply:
These columns were previously dropped from the User table when Device's trackable module was removed, I believe: https://gsa-tts.slack.com/archives/C0NGESUN5/p1649098117066389

I guess that had the distinction:
current_sign_in_at vs. last_sign_in_at

Reply:
Yep, they were unused (and duplicative of events), so I think it's fine to use events.  I don't mind the naming, I think it's clear and explicit 🙂

When someone signs in, they will create both a `sign_in_before_2fa` and `sign_in_after_2fa` event, but I'm not sure if that means the 2nd most recent event will likely always be within the last few minutes.  If it does, that's probably a decent enough reason to drop `sign_in_before_2fa` from the query?